core/src/test/java/org/apache/iceberg/TestScanSummary.java

```diff
@@ -44,8 +44,9 @@ public void testSnapshotTimeRangeValidation() {
         .appendFile(FILE_B) // data_bucket=1
         .commit();
 
+    // Busy wait for our click to tick forward.
     long t1 = System.currentTimeMillis();
-    while (t1 == t0) {
```

Comment:
I think the underlying problem is not the logic here, but the assumption that `t0` is equivalent to the last committed snapshot's timestamp. What about a fix like this?

```
--- a/core/src/test/java/org/apache/iceberg/TestScanSummary.java
+++ b/core/src/test/java/org/apache/iceberg/TestScanSummary.java
@@ -45,7 +45,7 @@ public class TestScanSummary extends TableTestBase {
         .commit();

     long t1 = System.currentTimeMillis();
-    while (t1 == t0) {
+    while (t1 <= table.currentSnapshot().timestampMillis()) {
       t1 = System.currentTimeMillis();
     }

@@ -56,7 +56,7 @@ public class TestScanSummary extends TableTestBase {
     long secondSnapshotId = table.currentSnapshot().snapshotId();

     long t2 = System.currentTimeMillis();
-    while (t2 == t1) {
+    while (t2 == table.currentSnapshot().timestampMillis()) {
       t2 = System.currentTimeMillis();
     }
```